### DISCUSSION TITLE
Available and Total quantity

### DISCUSSION THREAD

**Discussion by andreievg** [msupply-foundation/open-msupply#5807] [General]
**Total** is all stock at a facility.
**Available** is total minus anything reserved by shipments.

This discussion is a placeholder for a conversation to get a clear guidance on where and when we should show total and/or available, also how we communicate this to users and stakeholder.

Recent questions and discussion about this:

1 - What to show in item usage report, mSupply shows total, in omSupply looks like we are showing available (i think to align with SOH figures in other areas ?)
2 - We show total when selecting item in outbound shipment line edit, but none maybe available (so can't actually issue that item)

**Comment by mark-prins** [comment]
where I've got to currently:
* the item usage report is identifying stock that you have on hand and extrapolating months of cover etc based on that. given that the stock is still in store, even if allocated to a shipment, it makes sense to calculate and display total stock; if there is a need you can re-allocate as the stock is still physically in store
* the outbound shipment however is the point that you are allocating stock, and this should respect other allocations i.e. this should be using available primarily, though it could be helpful for the display to show total stock, as there may be a need to reallocate

**Comment by craigdrown** [comment]
First, if anyone changes anything like this from the way it has been done in OG, we need to have explained that at the time, and validated that with the OGers and also explained how future migration will work. 

Underlying Q is "what constitutes "stock on hand" - our historical understanding is that
1. total stock is the total in the warehouse.
2. difference between available and total is what is allocated (likely on packing benches or boxed waiting for transport)
3. ownership transfers when the invoice is confirmed (goods are shipped, total stock on hand is decreased) >> should use stock on hand (not available) for most reporting.

**Reply by Gwalior-84** [reply]
This makes good sense to me.  I'd also observe that most of the time, users would be reporting SoH for a much more general purpose (like to work out how much to order) rather than whether we have happened to ship a particular outbound shipment today.  So the actual shipped status of a particular outbound shipment should generally be 'noise' and we should be able to ignore it.

From that, I conclude that the only time we use / show _available stock_ instead of _total stock_ is when selecting stock for outbound shipments, where the user should only be allowed to pick from available stock.  Is there any other use case?
